packages/core/src/shared/extensions/ssh.ts

```diff
@@ -119,6 +123,26 @@ export class RemoteSshSettings extends Settings.define('remote.SSH', remoteSshTy
     }
 }
 
+export async function testSshConnection(
+    ProcessClass: typeof ChildProcess,
+    hostname: string,
+    sshPath: string,
+    user: string,
+    session: SSM.StartSessionResponse
+): Promise<void> {
+    try {
+        const env = { SESSION_ID: session.SessionId, STREAM_URL: session.StreamUrl, TOKEN: session.TokenValue }
+        await new ProcessClass(sshPath, ['-T', `${user}@${hostname}`, 'echo connected && exit']).run({
```

Comment:
not a blocker: for connecting to a Windows machine would we need a different test command? or does this command work on powershell?

Reply:
Since we directly invoke the sshPath I believe this should still work. However, once I add some tests for this function I will hopefully have a better idea.